eventstore_client.go

```diff
@@ -76,15 +92,139 @@ func getStats() (*stats, error) {
 		}
 	}
 
+	var parkedMessagesStatsResult []parkedMessagesStats
+	if enableParkedMessagesStats {
+		parkedMessagesStats, err := getSubscriptionParkedMessagesStats(subscriptionsStatsResult.result)
+		if err != nil {
+			log.WithError(err).Error("Error while getting parked messages for subscriptions.")
+		} else {
+			parkedMessagesStatsResult = *parkedMessagesStats
+		}
+	}
+
 	return &stats{
 		serverStatsResult.result,
 		gossipStatsResult.result,
 		projectionStatsResult.result,
 		infoResult.result,
 		subscriptionsStatsResult.result,
+		parkedMessagesStatsResult,
 	}, nil
 }
 
+func getSubscriptionParkedMessagesStats(subscriptions []byte) (*[]parkedMessagesStats, error) {
+	var result []parkedMessagesStats
+	jp.ArrayEach(subscriptions, func(jsonValue []byte, dataType jp.ValueType, offset int, err error) {
+		eventStreamID, _ := jp.GetString(jsonValue, "eventStreamId")
+		groupName, _ := jp.GetString(jsonValue, "groupName")
+
+		lastEventNumber, err := getParkedMessagesLastEventNumber(eventStreamID, groupName)
+
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"eventStreamId": eventStreamID,
+				"groupName":     groupName,
+				"err":           err,
+			}).Warn("Error while getting parked messages last event number.")
+		}
+
+		truncateBeforeValue, err := getParkedMessagesTruncateBeforeValue(eventStreamID, groupName, lastEventNumber)
+
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"eventStreamId": eventStreamID,
+				"groupName":     groupName,
+				"error":         err,
+			}).Warn("Error while getting parked messages truncate before value.")
+		}
+
+		totalNumberOfParkedMessages := lastEventNumber - truncateBeforeValue
+
+		var oldestParkedMessage float64
+		oldestMessageID := lastEventNumber - totalNumberOfParkedMessages
+		oldestParkedMessage, err = getOldestParkedMessageTimeInSeconds(eventStreamID, groupName, oldestMessageID)
+		if err != nil {
+			log.WithFields(logrus.Fields{
```

Comment:
If all messages have been properly replayed, the code for oldest parked message attempts to get a non-existent message from the stream and warning is logged here. This should be handled and the request should not be issued.

Reply:
https://github.com/marcinbudny/eventstore_exporter/pull/14/commits/d0297a4e6c7912947714122f5a80c937856b067b